src/gui/qgsadvanceddigitizingdockwidget.cpp

```diff
@@ -237,6 +240,22 @@ QgsAdvancedDigitizingDockWidget::QgsAdvancedDigitizingDockWidget( QgsMapCanvas *
   constructionModeToolButton->setMenu( constructionSettingsMenu );
   constructionModeToolButton->setObjectName( QStringLiteral( "ConstructionModeButton" ) );
 
+  // Tools
+  QMenu *toolsMenu = new QMenu( this );
+  const QStringList toolMetadataNames = QgsGui::instance()->advancedDigitizingToolsRegistry()->toolMetadataNames();
```

Comment:
If this is built on QgsAdvancedDigitizingDockWidget construction then it'll never react to plugin installs/changes. I'd move this logic to a slot connected to toolsMenu::aboutToShow, and populate the menu at time of display instead.

Reply:
@nyalldawson , good idea; I'm also adding signals to the register but here the about to show is a much better way to go.